### ISSUE TITLE
False positive confusing error for incorrect inherent impl

### ISSUE THREAD

**Issue by pczarn** [rust-lang/rust-analyzer#17682]
**rust-analyzer version**: 0.3.2037-standalone

**rustc version**: 1.79.0

**editor or extension**: VSCode

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**repository link (if public, optional)**: https://github.com/pczarn/cfg

**code snippet to reproduce**: 
```rust
struct FirstSets;

impl FirstSets {
    fn new(grammar: &Cfg) -> Self { FirstSets }
}

impl Cfg {
    fn first_sets(&self) -> FirstSets {
        FirstSets::new(self)
//                   ^ no function or associated item named `new` found for struct `first::FirstSets` in the current scope
    }
}
```

**Comment by Veykril** [comment]
![image](https://github.com/user-attachments/assets/bc83ef3a-500f-4986-86d2-e0dd52e90a41)
your snippet works fine for me, where does this trigger in the linked repo?

**Comment by pczarn** [comment]
This is the code: https://github.com/pczarn/cfg/blob/a0abf520fcd7256517953badb7596fda89f78b15/cfg-predict-sets/src/first.rs#L140

Happens as well with cargo build. Happens with the latest rust-analyzer and rustc 1.82. I have more errors in the repo, but they are okay.

**Comment by pczarn** [comment]
Happens in this PR: https://github.com/pczarn/cfg/pull/20/files#diff-28f4cffd648e23ce69126c2bf87c3f2c3273a3a85ac3c143aee8ed09ffdd9a47

**Comment by pczarn** [comment]
Happens as well with the most recent nightly.

**Comment by pczarn** [comment]
@Veykril The `Cfg` implementee must be imported from a foreign crate, and I'd expect one error message, not an additional second error message, which is confusing.

**Comment by Veykril** [comment]
Hmm, I only get one error here
![image](https://github.com/user-attachments/assets/a10b56ce-1afc-401c-9b9e-ea379e5e541a)
and that error is correct

**Comment by pczarn** [comment]
Ah! Sorry. Thanks. I overlooked an error in the fn impls for `FirstSets`. The fns do not compile at all.
